op/model.py

```diff
@@ -229,6 +231,33 @@ def _load(self):
         return self._backend.config_get()
 
 
+class Resources:
+    """Object representing resources for the charm.
+    """
+    def __init__(self, names, backend):
+        self._names = names
+        self._backend = backend
+        self._path = None
+
+    def fetch(self, name):
+        """Fetch the resource from the controller or store.
+
+        If successfully fetched, this returns a Path object to where the resource is stored
+        on disk, otherwise it returns None.
+        """
+        if name not in self._names:
+            raise RuntimeError(f'invalid resource name: {name}')
+        if self._path is None:
```

Comment:
Isn't this true for only 1 of the resources? eg, you could ask to fetch resource A and resource B, but this would cache the fetch for the first one and cause you to not download the second one?

Shouldn't there be a single Resource type if we want to do caching of the path, and have Resources index into them?